src/TAO/API/types/system/validate.cpp

```diff
@@ -81,15 +84,26 @@ namespace TAO
                         /* Set the register type */
                         jsonRet["type"]    = RegisterType(state.nType);
 
-                        /* If it is an object register than parse it to add the object_type */
+                        /* If it is an object register then parse it to add the object_type */
                         if(state.nType == TAO::Register::REGISTER::OBJECT)
                         {
-                        
+
+                            uint256_t hashGenesis = users->GetCallersGenesis(params);
+
                             /* parse object so that the data fields can be accessed */
-                            if(state.Parse())
+                            if(state.Parse()) {
+                                if(hashGenesis != 0) {
+                                    if(state.hashOwner == hashGenesis)
+                                        jsonRet["is_mine"] = true;
+                                    else
+                                        jsonRet["is_mine"] = false;
+                                }
```

Comment:
can you move this logic outside of the check to see if state.nType is OBJECT, because you can own any type of register not just object registers...